public/app/features/explore/Logs/LogsTableActiveFields.tsx

```diff
@@ -0,0 +1,98 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { DragDropContext, Draggable, DraggableProvided, Droppable, DropResult } from 'react-beautiful-dnd';
+
+import { GrafanaTheme2 } from '@grafana/data/src';
+import { useTheme2 } from '@grafana/ui/src';
+
+import { LogsTableEmptyFields } from './LogsTableEmptyFields';
+import { LogsTableNavField } from './LogsTableNavField';
+import { fieldNameMeta } from './LogsTableWrap';
+
+export function getLogsFieldsStyles(theme: GrafanaTheme2) {
+  return {
+    wrap: css({
+      marginTop: theme.spacing(1),
+      marginBottom: theme.spacing(1),
+      display: 'flex',
+    }),
+    columnWrapper: css({
+      marginBottom: theme.spacing(1.5),
+      // need some space or the outline of the checkbox is cut off
+      paddingLeft: theme.spacing(0.5),
+    }),
+  };
+}
+
+function sortLabels(labels: Record<string, fieldNameMeta>) {
+  return (a: string, b: string) => {
+    const la = labels[a];
+    const lb = labels[b];
+
+    // Sort by index
+    if (la.index != null && lb.index != null) {
+      return la.index - lb.index;
+    }
+
+    // otherwise do not sort
+    return 0;
+  };
+}
+
+export const LogsTableActiveFields = (props: {
+  labels: Record<string, fieldNameMeta>;
+  valueFilter: (value: string) => boolean;
+  toggleColumn: (columnName: string) => void;
+  reorderColumn: (sourceIndex: number, destinationIndex: number) => void;
+  id: string;
+}): JSX.Element => {
+  const { reorderColumn } = props;
+
+  const onDragEnd = (result: DropResult) => {
+    if (!result.destination || !reorderColumn) {
```

Comment:
```suggestion
    if (!result.destination) {
```
Shouldn't be needed, right?